File: kotlin/src/main/kotlin/HttpClient.kt
package com.vo

import java.time.ZonedDateTime

data class Response(val inAmount: Int, val outAmount: Int, val name: String, val timestamp: ZonedDateTime, val serial: String)

class HttpClient {
    fun fetch(url : String) = Response(10, 5, "foo", ZonedDateTime.now(), "xd13f0af")
}

File: ts/src/HttpClient.ts
export type Response = {
  inAmount: number;
  outAmount: number;
  name: string;
  timestamp: Date;
  serial: string;
};
export class HttpClient {
  fetch(_url: string) {
    return {
      inAmount: 10,
      outAmount: 5,
      name: "Foo",
      timestamp: new Date(),
      serial: "xd13f0af",
    };
  }
}
